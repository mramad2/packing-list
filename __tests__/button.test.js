import React from "react"
import { Button } from "../src/components/button"
import renderer from "react-test-renderer"

test("button has onPress fn", () => {
  let x = 0
  const buttonInstance = renderer.create(<Button onPress={() => x++} />).getInstance()
  expect(buttonInstance.props.onPress).toBeDefined()
  expect(x).toBe(0)
  buttonInstance.props.onPress()
  expect(x).toBe(1)
})

test("Button shape has not changed", () => {
  const snapshot = renderer.create(<Button />).toJSON()
  expect(snapshot).toMatchSnapshot()
})

test("Button bg is grey on clear prop", () => {
  const snapshotClear = renderer.create(<Button clear />).toJSON()
  expect(snapshotClear).toMatchSnapshot()
})
