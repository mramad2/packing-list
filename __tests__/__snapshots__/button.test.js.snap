// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button bg is grey on clear prop 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "gray",
      "borderRadius": 5,
      "justifyContent": "center",
      "marginLeft": 10,
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 0.9,
      "shadowRadius": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "margin": 5,
      }
    }
  />
</View>
`;

exports[`Button shape has not changed 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "green",
      "borderRadius": 5,
      "justifyContent": "center",
      "marginLeft": 10,
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 0.9,
      "shadowRadius": 5,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "margin": 5,
      }
    }
  />
</View>
`;
